File: src/main/java/com/baimsg/thread/DictionaryPolicy.java
package com.baimsg.thread;

import java.util.concurrent.RejectedExecutionHandler;
import java.util.concurrent.ThreadPoolExecutor;

/**
 * Create by Baimsg on 2021/11/19
 **/
public class DictionaryPolicy implements RejectedExecutionHandler {
    @Override
    public void rejectedExecution(Runnable r, ThreadPoolExecutor executor) {
        System.out.println(r+ "被拒绝执行");
        if (!executor.isShutdown()) {
            r.run();
        }
    }
}

File: src/main/java/com/baimsg/thread/DictionaryPolicy.kt
package com.baimsg.thread

import java.util.concurrent.RejectedExecutionHandler
import java.lang.Runnable
import java.util.concurrent.ThreadPoolExecutor

/**
 * Create by Baimsg on 2021/11/19
 */
class DictionaryPolicy : RejectedExecutionHandler {
    override fun rejectedExecution(r: Runnable, executor: ThreadPoolExecutor) {
        println(r.toString() + "被拒绝执行")
        if (!executor.isShutdown) {
            r.run()
        }
    }
}